frame/staking/src/lib.rs

```diff
@@ -611,21 +621,19 @@ impl<T: Config> StakingLedger<T> {
 			slash_era,
 			self,
 			slash_chunks_priority,
-			maybe_proportional,
+			Perquintill::one() - remaining_ratio,
```

Comment:
This should be safe math. Or, yet again, I want to complain that the refactor is more of step backwards, since `Option` is the prefect type to show a `maybe_it_is_proportional`, instead of some value being `0`.

Reply:
The intention was to unify both cases (proportional or not) under a  ratio,  proportional or not just a matter of different ratio, but since it is important to get clear of these cases this indeed may not be a good idea